packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -108,6 +110,18 @@ export const estimateFee = async (
   });
 };
 
+export const waitForTransaction = async (
+  network: Network,
+  senderAddress: string,
+  privateKey: string | Uint8Array,
+  txnHash: num.BigNumberish,
+  cairoVersion?: CairoVersion,
+): Promise<GetTransactionReceiptResponse> => {
+  const provider = getProvider(network);
+  const account = new Account(provider, senderAddress, privateKey, cairoVersion ?? '0');
+  return account.waitForTransaction(txnHash);
```

Comment:
Should you `await` here ?

Reply:
in terms of the functionality, it is not required, as the caller has to use await, 
or
we can add this to the rest of the same pattern function in 1 batch to standardise the pattern